cmd/cri-resmgr-agent-probe/main.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"net"
+	"time"
+
+	"google.golang.org/grpc"
+
+	agent_v1 "github.com/intel/cri-resource-manager/pkg/agent/api/v1"
+	"github.com/intel/cri-resource-manager/pkg/cri/resource-manager/sockets"
+	"github.com/intel/cri-resource-manager/pkg/log"
+)
+
+func main() {
+	socket := flag.String("agent-socket", sockets.ResourceManagerAgent, "Unix domain socket where agent is serving")
+
+	flag.Parse()
+
+	// Try to connect to agent
+	dialOpts := []grpc.DialOption{
+		grpc.WithInsecure(),
+		grpc.WithDialer(func(sock string, timeout time.Duration) (net.Conn, error) {
+			return net.Dial("unix", sock)
+		}),
+	}
+	conn, err := grpc.Dial(*socket, dialOpts...)
+	if err != nil {
+		log.Fatal("failed to connect to agent: %v", err)
+	}
+	cli := agent_v1.NewAgentClient(conn)
+
+	// Do health check
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	rpl, err := cli.HealthCheck(ctx, &agent_v1.HealthCheckRequest{})
+	if err != nil {
```

Comment:
So just that I understand/we explicitly spell out the conventions here:
  - a `nil` error implies an `OK` `Status` in the reply (so it is safe to leave that unchecked)
  - a non-`nil` error with a non-`nil` `Status` implies an agent-level error (possibly with further information in `Status`  about what is wrong ?)
- a non-`nil` error with a `nil` reply implies a transport/communication level error

Reply:
Good questions @klihub. The PR was quickly patched together and the status field could also be dropped. 

The status field in the reply is just for informative purposes, that could be used to carry more detailed inforation about the state. Thus, the logic would be just like you described (we just don't really use the field for anything, yet). A non-healthy state always causes an error response. In practice, a non-nil rsp error with nil status would imply a "lower-level" error as you described, and similarly, a non-nil rsp error with a non-empty status would imply that the agent was able to respond but it is not functioning properly (status would carry more detailed info about the problems). And, a nil rsp error implies an healthy state, but, even in this case we could carry some information in the status field.